spec/support/test_helper.rb

```diff
@@ -1,24 +1,32 @@
 module DfE
   module Bigquery
     module TestHelper
-      include ::BigqueryStubs
-
-      def stub_bigquery_application_metrics_request(rows: nil, job_complete: true, page_token: nil)
+      def stub_bigquery_application_metrics_request(rows: nil, job_complete: true, page_token: nil, result: true)
         bigquery_client = instance_double(Google::Apis::BigqueryV2::BigqueryService)
         allow(DfE::Bigquery).to receive(:client).and_return(bigquery_client)
-        allow(bigquery_client).to receive(:query_job).and_return(stubbed_bigquery_application_metrics_response(rows:, job_complete:, page_token:))
+        allow(bigquery_client).to receive(:query_job).and_return(stubbed_bigquery_application_metrics_response(rows:, job_complete:, page_token:, result:))
       end
 
-      def stub_bigquery_application_metrics_by_provider_request(rows: nil, job_complete: true, page_token: nil)
+      def stub_bigquery_application_metrics_by_provider_request(rows: nil, job_complete: true, page_token: nil, result: true)
         bigquery_client = instance_double(Google::Apis::BigqueryV2::BigqueryService)
         allow(DfE::Bigquery).to receive(:client).and_return(bigquery_client)
-        allow(bigquery_client).to receive(:query_job).and_return(stubbed_bigquery_application_metrics_by_provider_response(rows:, job_complete:, page_token:))
+        allow(bigquery_client).to receive(:query_job).and_return(stubbed_bigquery_application_metrics_by_provider_response(rows:, job_complete:, page_token:, result:))
       end
 
-      def stubbed_bigquery_application_metrics_response(rows: nil, job_complete: nil, page_token: nil)
-        return stub_response(rows:, job_complete:, page_token:) if rows
+      # @param row [nil|Row|'nil']
+      #   When nil, it returns a default response
+      #   When Row it returns the values defined in the Row
+      #   When 'nil' it returns literal nil
+      # @param job_complete [Boolean] Is the response populated with the query result?
+      # @param page_token [String] If present, there is a next page that must be fetched
+      # @param result [Boolean] Does the response return values?
+      #
+      def stubbed_bigquery_application_metrics_response(rows: nil, job_complete: nil, page_token: nil, result: true)
+        return BigqueryStubs.stub_response(rows:, job_complete:, page_token:, result:) unless result
+
+        return BigqueryStubs.stub_response(rows:, job_complete:, page_token:, result:) if rows
 
-        stub_response(rows: [[
+        BigqueryStubs.stub_response(rows: [[
```

Comment:
I've made `stub_response` a module method becuase it's such a genericly named method it could clash with other methods by the same name.